### ISSUE TITLE
Perfect rollup is not perfect (maybe due to multi-value dimension?) 

### ISSUE THREAD

**Issue by gulinfu** [apache/druid#10238]
### Affected Version

0.18.1

### Description
ingestion spec:
```
{
  "type": "index_parallel",
  "spec": {
    "dataSchema": {
      "dimensionsSpec": {
        "dimensions": [
          "dimensionA",
          "dimensionB",
          "dimensionC",
          "dimensionD",
          "dimensionE",
          "dimensionF"
        ]
      },
      "metricsSpec": [
        {
          "fieldName": "firstPartyCookie",
          "name": "1stPartyCookie_hll",
          "type": "HLLSketchBuild"
        },
        {
          "fieldName": "thirdPartyCookie",
          "name": "3rdPartyCookie_hll",
          "type": "HLLSketchBuild"
        },
        {
          "fieldName": "viewCount",
          "name": "viewCount",
          "type": "longSum"
        }
      ],
      "granularitySpec": {
        "intervals": [
          "2020-08-04T15:00:00.000Z/2020-08-04T16:00:00.000Z"
        ],
        "segmentGranularity": "hour",
        "queryGranularity": "hour",
        "type": "uniform"
      },
      "timestampSpec": {
        "column": "timestamp",
        "format": "auto"
      },
      "dataSource": "xxx_production"
    },
    "ioConfig": {
      "inputSource": {
        "prefixes": [
          "s3://xxx/2020-08/04/15"
        ],
        "type": "s3"
      },
      "type": "index_parallel",
      "inputFormat": {
        "listDelimiter": "\u0001",
        "columns": [
          "timestamp",
          "dimensionA",
          "dimensionB",
          "dimensionC",
          "dimensionD",
          "dimensionE",
          "dimensionF"
          "firstPartyCookie",
          "thirdPartyCookie",
          "viewCount"
        ],
        "type": "csv"
      }
    },
    "tuningConfig": {
      "forceGuaranteedRollup": "true",
      "partitionsSpec": {
        "numShards": 1,
        "type": "hashed"
      },
      "maxNumConcurrentSubTasks": 7,
      "type": "index_parallel"
    }
  }
}
```

When I do some SQL `select *` query on the ingested druid data, I realize there are some rows with the same dimensions, which means they are not perfectly rolled up.

To validate my assumption, I tried to reindex the existing data source using druid inputSource, and the output segment is about 50% size and 50% number of rows compared to the origin. I did the same SQL query, and I don't see rows with the same dimensions this time. I think it's perfectly rolled up now after reindexing.

reindex spec:
```
{
  "type": "index_parallel",
  "spec": {
    "dataSchema": {
      "dataSource": "xxx_rollup_reindex",
      "timestampSpec": {
        "column": "timestamp",
        "format": "auto"
      },
      "dimensionsSpec": {
        "dimensions": [
          "dimensionA",
          "dimensionB",
          "dimensionC",
          "dimensionD",
          "dimensionE",
          "dimensionF"
        ]
      },
      "metricsSpec": [
        {
          "fieldName": "1stPartyCookie_hll",
          "name": "1stPartyCookie_hll",
          "type": "HLLSketchMerge"
        },
        {
          "fieldName": "3rdPartyCookie_hll",
          "name": "3rdPartyCookie_hll",
          "type": "HLLSketchMerge"
        },
        {
          "fieldName": "viewCount",
          "name": "viewCount",
          "type": "longSum"
        }
      ],
      "granularitySpec": {
        "type": "uniform",
        "segmentGranularity": "hour",
        "queryGranularity": "hour",
        "intervals": [
          "2020-08-04T15:00:00.000Z/2020-08-04T16:00:00.000Z"
        ]
      }
    },
    "ioConfig": {
      "type": "index_parallel",
      "inputSource": {
        "type": "druid",
        "dataSource": "xxx_production",
        "interval": "2020-08-04T15:00:00.000Z/2020-08-04T16:00:00.000Z"
      }
    },
    "tuningConfig": {
      "type": "index_parallel",
      "maxNumConcurrentSubTasks": 1,
      "forceGuaranteedRollup": "true",
      "partitionsSpec": {
        "numShards": 1,
        "type": "hashed"
      }
    }
  }
}
```

One of my dimension is a multi-value dimension, maybe that's the reason? But still, we would expect "forceGuaranteedRollup": "true" to give me a perfect rollup anyway.

Thanks!

**Comment by clintropolis** [comment]
Hi @gulinfu,
An issue with perfect roll-up of multi-value string dimensions was fixed by #9742, added in 0.19. Any chance you could test to see if this issue still occurs in this version?

**Comment by gulinfu** [comment]
@clintropolis 
Thanks for the info! I'll check that out.
BTW 0.19 middleManagers are supposed to work well with other components on 0.18.1 right?

**Comment by clintropolis** [comment]
I think this should be ok to mix versions, nothing is sticking out in the release notes at least.

**Comment by gulinfu** [comment]
validated - it's fixed on 0.19.0
thank you!
